util/multierror.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"strings"
+)
+
+type MultiError map[string]struct{}
+
+func NewMultiError(errs ...string) MultiError {
```

Comment:
Usually, we will use Error interface for error, but here use string instead. This is fine for now for the specific need. However, it will be good to improve this later, to make the MultiError support Error args instead of a raw string because you can use err.Error() to get the error message.

Reply:
Also, I would prefer using error.go instead of a specific multierror.go, because it will be much extensible in the future, but this is quite minor now :slightly_smiling_face:

Reply:
Yes, the naming is a bit of weird. In order to do a fast deduplication of errors, `map[string]error` is preferred, but I'm not quiet sure `ERRORREPLICASCHEDULEXXX` should string or error for now...
Maybe we should write our own `multierror` package following errors interface for our use case and replace current impl with it. Let me think about it more.

Reply:
yeah, I know you want to use map for set purpose.